Extensions/AboutUsParser.js

```diff
@@ -0,0 +1,325 @@
+ */
+async function createFilesArray(files) {
+    let mappedFiles = [];
+    files.map(file => {
+        file = file.split('/');
+        mappedFiles.push({ name: file[file.length - 1], path: Path.join(...file), language: file[file.length - 2], folder: file[file.length - 3] });
+    });
+    return mappedFiles;
+}
+
+/** Parses given string by separators and empty lines into an array of parsed sections
+ * 
+ * @param {String} string text to be parsed
+ * @param {Function} data callback function holding data
+ * @returns {Array} 
+ */
+function parseString(string) {
+    let sections = string.split(C.separator); // We split the string by our main separator into individual sections
+    let parsedSections = [];
+
+    sections.map(section => { // We iterate through every parsed section
+        let para = section.split('\r\n').filter(sentence => sentence.trim() ? true : false); // Splitting sections into paragraphs by new line and removing empty lines
+        parsedSections.push({ title: para[0], short: para[1], long: para.slice(2,para.length) }); // Pushing a parsed section object into an array
+    });
+
+    return parsedSections; // return array of parsed sections as a callback
+}
+
+/** Formats and appends HTML code from an Array of objects
+ * 
+ * @param {Array} array Array of objects that can be mapped into `new Paragraph()` 
+ * @returns {String} Formatted html string
+ */
+function getStringFromArray(array) {
+    let str_builder = '';
+    for (var par of array)
+        str_builder += new Paragraph(par).getDefaultHtmlCode();
+    
+    return str_builder;
+}
+
+/** Creates a new component for handlebars in the given directory with the given content
+ * 
+ * @param {String} Finalpath Path to directory where to create the component
+ * @param {String} name File name of the component
+ * @param {String} content String that will be written inside the component
+ * @param {String} language 
+ * @param {Boolean} append_file Whether to append or overwrite already existing file 
+ * @param {Boolean} nestedComponent Whether the component is nested or not
+ * @param {String} fileName Name of the component file to be created
+ */
+function createComponent(Finalpath = null, name, content, language, append_file = true, nestedComponent = true, fileName = null) {
+    fileName = fileName ? fileName : C.componentName;
+    Finalpath = Finalpath ? Finalpath : C.ComponentsPath;
+
+    const component = nestedComponent ? `:^) ${name} :::\r\n` : '';
+    let path = Path.join(Finalpath, language, `${fileName}.html`);
+
+    writeFile(path, `${component}${content}`, append_file);
+    console.log(`Successfully created component ${component},\r\nAppend File Mode: ${append_file}`);
+}
+
+/** Asynchronously writes a string to a file
+ * 
+ * @param {String} path Path of the file
+ * @param {String} string String to write into the file
+ * @param {Boolean} appendFile 
+ * @returns {void}
+ */
+function writeFile(path, string, appendFile) {
+    try {
```

Comment:
In case someone ducks up so that they know where and what happened and when and if they go through the code they will figure out where they went wrong.